clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/InternalLocalBootstrappingVeniceChangelogConsumer.java

```diff
@@ -175,7 +175,8 @@ private Function<String, Boolean> functionToCheckWhetherStorageEngineShouldBeKep
             : new VeniceChangeCoordinate(topicPartition.getPubSubTopic().getName(), earliestOffset, partition);
 
         // If earliest offset is larger than the local, we should just bootstrap from beginning.
-        if (earliestCheckpoint != null && earliestCheckpoint.comparePosition(localCheckpoint) > -1) {
+        if (earliestCheckpoint != null
```

Comment:
Could you add [cdc] tag as well? IIUC this is classic CDC implementation not the DVRT based solution.